src/middleware/persist.ts

```diff
@@ -5,19 +5,15 @@ import {
   StoreMutatorIdentifier,
 } from '../vanilla'
 
-type DeepPartial<T> = {
-  [P in keyof T]?: DeepPartial<T[P]>
-}
-
 export type StateStorage = {
   getItem: (name: string) => string | null | Promise<string | null>
   setItem: (name: string, value: string) => void | Promise<void>
   removeItem: (name: string) => void | Promise<void>
 }
 
-type StorageValue<S> = { state: DeepPartial<S>; version?: number }
+type StorageValue<S> = { state: S; version?: number }
```

Comment:
There's no reason for this to be `DeepPartial<S>` so I made it `S`.

Reply:
Not sure to understand what you mean by "the types are lying" 🤔 
Does it work with `partialize`?

Reply:
> Not sure to understand what you mean by "the types are lying" 🤔

What I mean is types are not in alignment with the runtime. In runtime we're storing `{ state: S, version?: number }` in the storage but the types say we're storing `{ state: DeepPartial<S>, version?: number }` hence they're lying.

> Does it work with partialize?

Yes it does, I'll add a test case